Bring up business structure prototype with tests

matejak opened the pull request:
This WIP PR features a prototype of business logic with unit tests. There is no documentation, but refer to the unit tests, they are supposed to correspond to some of the use cases.

Featured functionality:

- There can be activity requests and cancellation requests (that can cancel activities)
- A storage class stores requests and it is able to return requests according to their type.
- Request pool takes requests from storage and exposes time-relevant requests that have not been cancelled.
- Requests filtered by the pool can be grouped together (partitioned).

matejak committed:
Bring up business structure prototype with tests

matejak committed:
Add ability to ignore conflicting requests

xmalmed commented on src/iwant_bot/business.py:
```diff
@@ -0,0 +1,105 @@
+        elif isinstance(request, ICancelRequest):
+            destination = self._requests["cancel"]
+        else:
+            raise ValueError("Can't store requests of type {type(request)}.")
```
f ... (f"Can't store requests of type {type(request)}.")

matejak committed:
Fix the exception message of MemoryStorage + add tests.

matejak committed:
Divide business logic into separate files

matejak committed:
Improved RequestStorage retreival mechanism.

Enable filtering of requests retreival according to the
request's associated activity.

matejak committed:
Add the essential docstrings

matejak committed:
Remove the cancellation requests mechanics.

This was a dead-end as the cancellation mechanics will be handled differently.

matejak commented:
I have divided the business logic stuff into multiple files and did some adjustments upon which we have agreed on the mentoring session.
I propose that this PR is merged as-is, so that others can start their own PRs from the merged master without delays. I am away and probably also off the grid for the weekend.

matejak committed:
Fix the code to conform to PEP8

kandski committed:
Add simple pairing

Simple pairing prototype which creates pairs
from every active requests in storage.
No options or preferences accepted.

kandski committed:
Fix the code to comply with PEP8
